src/Interpreters/SystemLog.cpp

```diff
@@ -462,41 +457,33 @@ void SystemLog<LogElement>::savingThreadFunction()
 {
     setThreadName("SystemLogFlush");
 
-    std::vector<LogElement> to_flush;
-    bool exit_this_thread = false;
-    while (!exit_this_thread)
+    while (true)
     {
         try
         {
-            // The end index (exclusive, like std end()) of the messages we are
-            // going to flush.
-            uint64_t to_flush_end = 0;
-            // Should we prepare table even if there are no new messages.
-            bool should_prepare_tables_anyway = false;
-
-            to_flush_end = queue->pop(to_flush, should_prepare_tables_anyway, exit_this_thread);
+            auto result = queue->pop();
 
-            if (to_flush.empty())
+            if (result.is_shutdown)
             {
-                if (should_prepare_tables_anyway)
-                {
-                    prepareTable();
-                    LOG_TRACE(log, "Table created (force)");
+                LOG_TRACE(log, "Terminating");
```

Comment:
Should we include the table name here? Or does the logger name contain it already?

Reply:
Yes, `log` are pretty good there, like `SystemLog (system.query_log)`.